content/post/087-js-ipfs-0.42.md

```diff
@@ -0,0 +1,196 @@
+
+```console
+$ npm install -g ipfs@next
+or
+$ npm install ipfs-http-client@next
+```
+
+This will download and install a version of `js-ipfs` and/or `js-ipfs-http-client` created from the last successful [build](https://travis-ci.com/github/ipfs/js-ipfs/branches) of master with no GitHub URLs in the production dependency tree.
+
+## 🛳️ Updated Docker containers
+
+Publishing to Docker Hub has been restored so you can get a container running the latest and greatest `js-ipfs` with:
+
+```console
+$ docker pull ipfs/js-ipfs
+```
+
+You can also pull pre-releases at any time with:
+
+```console
+$ docker pull ipfs/js-ipfs:next
+```
+
+Check out all the releases at: https://hub.docker.com/r/ipfs/js-ipfs
+
+## ✨New features
+
+The [MFS][] CLI now supports nanosecond mtimes, use it with the following commands:
+
+```console
+$ jsipfs mkdir /directory --mtime 1000 --mtime-nsecs 5
+```
+
+`--mode` and `--mtime` are now also supported when piping files to `jsipfs add`:
+
+```console
+$ cat ./file.txt | jsipfs add --mode 0500
+```
+
+## 🔨 Breaking changes
+
+- [HAMT](https://en.wikipedia.org/wiki/Hash_array_mapped_trie) sharded directories are now reported as having a type of `'directory'` from `ipfs.files.stat` in line with `go-ipfs`
+- When specifying metadata and `--raw-leaves` to `ipfs.add`, if the file is small enough to fit into one block, `--raw-leaves` is ignored and a [UnixFS][] entry is created (previously the returned [CID][] would have a codec of `raw`)
+
+## 🦟 Bugs fixed
+
+- WebRTC configuration is now passed to the libp2p WebRTC transport correctly [ipfs/js-ipfs#2966](https://github.com/ipfs/js-ipfs/pull/2966)
+- Unknown CLI commands now cause the process to exit with a non-zero exit code [ipfs/js-ipfs#2916](https://github.com/ipfs/js-ipfs/pull/2916)
+- Running Prometheus metrics in Docker containers no longer prevent the container from starting [ipfs/js-ipfs#2952](https://github.com/ipfs/js-ipfs/issues/2952)
+
+## 💍 One PR to rule them all
+
+A common observation made about IPFS development is that there are so many repos to gain knowledge of, it’s hard to know how everything fits together and where a problem might originate, all of which makes it hard to start contributing.
+
+Contributors will sometimes spend precious time crafting one PR, only to be told they need to make additional PRs to repos they didn't even know existed in order to land their initial change. This often ends up being too much of a barrier for developers who have external time pressures we aren’t aware of.
+
+Features must be staged across multiple releases from multiple repos, any of which have the possibility of blocking other work streams if human error bugs are introduced, or perhaps a piece of work was not as fully specified as it needed to be.
```

Comment:
Don't quite parse this. How does not specifying your work fit in? Maybe break into two sentences?

Reply:
I've reworded that part, hopefully it's a bit clearer now!